snapshot-test/src/main/java/com/hazelcast/jet/tests/snapshot/SnapshotTradeProducer.java

```diff
@@ -43,15 +51,38 @@ public void close() {
     }
 
     public void produce(String topic, int countPerTicker) {
-        for (long i = 0; i < Long.MAX_VALUE; i++) {
+        futureList = new ArrayList<>(countPerTicker);
+        long item = 0;
+        int itemCount = countPerTicker;
+        while (true) {
+            long sent = sendItems(topic, item, itemCount);
+            if (sent != itemCount) {
```

Comment:
we should also check and respond to `sent > itemCount` in produce method - if this happens due to any issue (e.g. after some incorrect adjustment in `sendItems` method) it can be infinite.